### ISSUE TITLE
[Bug]autoslim, use splitted checkpoint to  test subnet, accuracy is wrong

### ISSUE THREAD

**Issue by priscillapan** [open-mmlab/mmrazor#66]
### Describe the bug
I followed the 5-step instruction to use autoslim, use the default config file.
In step 3, I retrain the subnet at 199M Flops, whose top-1 accuracy is 70.13%.
then I Split checkpoint like step 4.

In step 5, if I use the splitted checkpoint, the accuracy is completely wrong. But if I use the checkpoint in step 3, the accuracy is right. 
The log is as follows:

**The logs for splitted checkpoint**

`/usr/local/lib/python3.8/dist-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See
https://pytorch.org/docs/stable/distributed.html#launch-utility for
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
*****************************************
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/split_mbv2_subnet_199mflops/checkpoint_1.pth
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
./tools/mmcls/test_mmcls.py:134: UserWarning: Class names are not saved in the checkpoint's meta data, use imagenet by default.
  warnings.warn('Class names are not saved in the checkpoint\'s '
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 50000/50000, 1094.2 task/s, elapsed: 46s, ETA:     0s
accuracy_top-1 : 0.10

accuracy_top-5 : 0.50`

**The logs for subnet trained checkpoint**

`/usr/local/lib/python3.8/dist-packages/torch/distributed/launch.py:178: FutureWarning: The module torch.distributed.launch is deprecated
and will be removed in future. Use torchrun.
Note that --use_env is set by default in torchrun.
If your script expects `--local_rank` argument to be set, please
change it to read from `os.environ['LOCAL_RANK']` instead. See
https://pytorch.org/docs/stable/distributed.html#launch-utility for
further instructions

  warnings.warn(
WARNING:torch.distributed.run:
*****************************************
Setting OMP_NUM_THREADS environment variable for each process to be 1 in default, to avoid your system being overloaded, please further tune the variable for optimal performance in your application as needed.
*****************************************
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
load checkpoint from local path: /root/work/mmrazor/work_dirs/autoslim_mbv2_subnet_8xb256_in1k/epoch_300.pth
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 50000/50000, 1085.0 task/s, elapsed: 46s, ETA:     0s
accuracy_top-1 : 70.13

accuracy_top-5 : 89.47`

**Comment by HIT-cwh** [comment]
Thanks for your issue. Is the order of the ``channel-cfgs`` in step 4 the same as that of ``algorithm.channel_cfg`` in step 3?

**Comment by priscillapan** [comment]
> Thanks for your issue. Is the order of the `channel-cfgs` in step 4 the same as that of `algorithm.channel_cfg` in step 3?

Yes.  the `channel-cfgs` in step 4 is the same as  that of `algorithm.channel_cfg` in step 3, the same file.

**Comment by HIT-cwh** [comment]
I'm a little confused. Maybe you do not need to split checkpoint when you only retrain one subnet('algorithm.channel_cfg' is only 1 file). Only when you retrain multiple subnets together (just like autoslim), you need to split the checkpoint before you test subnets with different flops.

By the way, experiments show that replacing ``LabelSmoothLoss`` with ``CrossEntropyLoss`` may lead to better results. And we will fix it later.

**Comment by priscillapan** [comment]
When i only retrain one subnet， i dont need to split checkpoint, I knew it when I tried to retrain 3 subnets later.
But when the first time i used it, i did not know. you should fix the split step or include a detailed explanation.
